npm/vue/src/index.ts

```diff
@@ -427,12 +426,29 @@ export function mount (componentOptions: any, options: any = {}) {
         logInstance.end()
       }
 
-      // by returning undefined we keep the previous subject
-      // which is the mounted component
-      return {
-        wrapper,
-        component: wrapper.vm,
-      }
+      return new Proxy(Object.create({}), {
+        get (obj, prop) {
+          if (Reflect.get(wrapper, prop)) {
+            // @ts-expect-error - internal API
+            Cypress.utils.throwErrByPath('mount.vue_yielded_value')
+          }
+
+          if (prop === 'then') {
+            return {
+              wrapper,
+              component: wrapper.vm,
+            }
+          }
+
+          if (prop === 'wrapper') {
+            return wrapper
+          }
+
+          if (prop === 'component') {
+            return wrapper.vm
+          }
+        },
+      })
```

Comment:
```suggestion
      const returnVal = {
        wrapper,
        component: wrapper.vm
      }
      
      return new Proxy(Object.create(returnVal), {
        get (obj, prop) {
        // throw an error if it looks like the caller is trying to call a method on the VueWrapper that was originally returned
          if (Reflect.get(wrapper, prop)) {
            // @ts-expect-error - internal API
            Cypress.utils.throwErrByPath('mount.vue_yielded_value')
          }

          return Reflect.get(obj, prop)
        },
      })
```

This implementation would handle adding a new property to the `returnVal` without having to also add an `if` clause in the proxy for it.  This is still just a suggestion.

Reply:
This makes sense. The previous behavior did not allow for `cy.mount(...).its('component')`, you suggested implementation fixes this.